### PR TITLE
TN-1824 migration to patch incorrect QB (wrong RP) on several QNRs.

### PR DESCRIPTION
**Pull request by pbugni**
The run time is down to about 30 seconds.

### CODE REVIEW

File: portal/migrations/versions/55469bdd181f_.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        rp3_name = "IRONMAN_v3" + qb.name[len('IRONMAN'):]
+        rp3_qb = QuestionnaireBank.query.filter_by(name=rp3_name).one()
+        assert (rp3_qb.research_protocol_id == rp3_id)
+        qb_map[qb.id] = rp3_qb.id
+    return qb_map
+
+
+def upgrade():
+    # Tricky task to locate the errors.  One time expensive process,
+    # loop through patients identifying problems.
```
Comment (ivan-c) [inline portal/migrations/versions/55469bdd181f_.py]: >Per email "RE: 170-99-003" 2018-01-17 to truenth-dev


If it's a specific problem, can you give a brief description? Not sure the exact issue after looking at referenced email...

Comment (pbugni) [reply portal/migrations/versions/55469bdd181f_.py]: Added detailed description to the jira issue.

File: portal/migrations/versions/55469bdd181f_.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                    expected_qb = QuestionnaireBank.query.get(
+                        qb_status._current_indef.qb_id)
+
+                if recorded_qb != expected_qb:
+                    replacement = qb_map.get(recorded_qb.id)
+                    if replacement == expected_qb.id:
+                        # retain the change request, as altering it inline
+                        # interrupts the iteration
+                        fix_map[qnr.id] = expected_qb.id
+                        print(
```
Comment (ivan-c) [inline portal/migrations/versions/55469bdd181f_.py]: Could you make this an audit log entry instead?

Comment (pbugni) [reply portal/migrations/versions/55469bdd181f_.py]: funny you should ask!  i had the same thought - see latest commit.

**Pull request merged**